### PR TITLE
Archive: Move logic to separate class

### LINKED ISSUE #9296: Archive: Move code to separate class
## Task Description

As discussed, all the custom archive page handling could live in its own class, reducing complexity of the `Story_Post_Type` class

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Move story archive logic into another class.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9296

### CODE REVIEW

File: includes/Story_Archive.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	 *
+	 * @since 1.13.0
+	 *
+	 * @param string[]|mixed $post_states An array of post display states.
+	 * @param WP_Post        $post        The current post object.
+	 *
+	 * @return string[]|mixed Filtered post display states.
+	 */
+	public function filter_display_post_states( $post_states, WP_Post $post ) {
+		if ( ! is_array( $post_states ) ) {
```
Comment (spacedmonkey) [inline includes/Story_Archive.php]:
```suggestion
	public function filter_display_post_states( $post_states, $post ) {
		if ( ! is_array( $post_states ) || ! $post ) {
```

**Comment by swissspidy** [comment]
Merging in main should fix the tests

File: includes/Story_Archive.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	 * @since 1.13.0
+	 *
+	 * @return void
+	 */
+	public function update_archive_setting() {
+		if ( ! defined( '\WPCOM_IS_VIP_ENV' ) || false === \WPCOM_IS_VIP_ENV ) {
+			$this->story_post_type->unregister_post_type();
+			$this->story_post_type->register_post_type();
+			flush_rewrite_rules( false );
+		}
```
Comment (swissspidy) [inline includes/Story_Archive.php]: I wonder if this shouldn't have been like this all along 🤔 

```suggestion
		$this->story_post_type->unregister_post_type();
		$this->story_post_type->register_post_type();
			
		if ( ! defined( '\WPCOM_IS_VIP_ENV' ) || false === \WPCOM_IS_VIP_ENV ) {
			flush_rewrite_rules( false );
		}
```

**Review by swissspidy — APPROVED** [review]
Just one question while at it

File: includes/Story_Archive.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Story_Archive.php]: Why not.

**Pull request merged**